harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -210,6 +364,26 @@ def _set_data_loaders(self) -> None:
                     )
                 self.validation_loader = validation_data
 
+    def _run(self):
+        min_checkpoint_period = self.env.experiment_config.get_min_checkpoint_period()
+        min_validation_period = self.env.experiment_config.get_min_validation_period()
+
+        for op in self.core_context.searcher.operations(core.SearcherMode.ChiefOnly):
+            searcher_unit = self.core_context.searcher.get_configured_units()
+            train_unit = None
```

Comment:
If you have `train_unit = None` up here you'll need an `if train_unit is None: raise ...` at the bottom of these if statements I think.